packages/wrangler/src/deploy/deploy.ts

```diff
@@ -386,12 +386,38 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 
 	const compatibilityFlags =
 		props.compatibilityFlags ?? config.compatibility_flags;
-	const nodejsCompat = compatibilityFlags.includes("nodejs_compat");
+
+	const nodejsCompatV2 = compatibilityFlags.includes(
+		"experimental:nodejs_compat_v2"
+	);
+	const nodejsCompatV2NotExperimental =
+		compatibilityFlags.includes("nodejs_compat_v2");
+	if (nodejsCompatV2) {
+		// strip the "experimental:" prefix because workerd doesn't understand it yet.
+		compatibilityFlags[
+			compatibilityFlags.indexOf("experimental:nodejs_compat_v2")
+		] = "nodejs_compat_v2";
+	}
+	// nodejsCompatV2 supersedes nodejsCompat, so disable nodejsCompat if nodejsCompatV2 is enabled
+	const nodejsCompat =
+		!nodejsCompatV2 ?? compatibilityFlags.includes("nodejs_compat");
```

Comment:
what I wanted to do was to support the case when both flags are present in the compat flags list.

I just wanted to be nice and I also didn't want to add more code to error when both flags are present, since `nodejs_compat_v2` will be rolled into `nodejs_compat` eventually.

I don't feel very strongly about this though.

Reply:
I understand that but this approach assumes the user implicitly understands the behaviour when both flags are enabled. It is kind of intuitive but...?

Reply:
I'm going to make this a hard error for now. We can relax it if we find this is too restrictive and annoying.